File: test-suite/src/test/java/io/micronaut/docs/server/response/ProducesController.java
package io.micronaut.docs.server.response;

//tag::imports[]
import io.micronaut.context.annotation.Requires;
import io.micronaut.http.HttpResponse;
import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Produces;
//end::imports[]

@Requires(property = "spec.name", value = "producesspec")
//tag::clazz[]
@Controller("/produces")
public class ProducesController {

    @Get // <1>
    public HttpResponse index() {
        return HttpResponse.ok().body("{\"msg\":\"This is JSON\"}");
    }

    @Produces(MediaType.TEXT_HTML)
    @Get("/html") // <2>
    public String html() {
        return "<html><title><h1>HTML</h1></title><body></body></html>";
    }

    @Get(value = "/xml", produces = MediaType.TEXT_XML) // <3>
    public String xml() {
        return "<html><title><h1>XML</h1></title><body></body></html>";
    }
}
//end::clazz[]

File: test-suite-python/src/test/python/micronaut/docs/server/response/ProducesController.py
# tag::imports[]
from micronaut.context.annotation import Requires
from micronaut.http import HttpResponse, MediaType
from micronaut.http.annotation import Controller, Get, Produces
# end::imports[]


@Requires(property="spec.name", value="producesspec")
# tag::clazz[]
@Controller("/produces")
class ProducesController:

    @Get  # <1>
    def index(self) -> HttpResponse:
        return HttpResponse.ok().body("{\"msg\":\"This is JSON\"}")

    @Produces(MediaType.TEXT_HTML)
    @Get("/html")  # <2>
    def html(self) -> str:
        return "<html><title><h1>HTML</h1></title><body></body></html>"

    @Get(value="/xml", produces=MediaType.TEXT_XML)  # <3>
    def xml(self) -> str:
        return "<html><title><h1>XML</h1></title><body></body></html>"
# end::clazz[]
